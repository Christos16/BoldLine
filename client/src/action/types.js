export const GET_PRODUCTS = 'GET_PRODUCTS';
export const GET_PRODUCT = 'GET_PRODUCT';
export const TOGGLE_CART_HIDDEN = 'TOGGLE_CART_HIDDEN';
export const ADD_ITEM = 'ADD_ITEM';
export const CLEAR_ITEM_FROM_CART = 'CLEAR_ITEM_FROM_CART';
export const REMOVE_ITEM = 'REMOVE_ITEM';
export const ADD_QUANTITY = 'ADD_QUANTITY';
export const DECREASE_QUANTITY = 'DECREASE_QUANTITY';
export const GET_CATEGORY = 'GET_CATEGORY';
export const SUCCESS_BUY = 'SUCCESS_BUY';
