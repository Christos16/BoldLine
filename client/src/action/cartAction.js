import {
  TOGGLE_CART_HIDDEN,
  ADD_ITEM,
  CLEAR_ITEM_FROM_CART,
  REMOVE_ITEM,
  ADD_QUANTITY,
  DECREASE_QUANTITY
} from './types';

export const toggleCartHidden = () => ({
  type: TOGGLE_CART_HIDDEN
});

export const addItem = item => ({
  type: ADD_ITEM,
  payload: item
});

export const clearItemFromCart = item => ({
  type: CLEAR_ITEM_FROM_CART,
  payload: item
});

export const removeItem = id => ({
  type: REMOVE_ITEM,
  payload: id
});

export const increaseQuantity = id => ({
  type: ADD_QUANTITY,
  payload: id
});

export const decreaseQuantity = id => ({
  type: DECREASE_QUANTITY,
  payload: id
});
