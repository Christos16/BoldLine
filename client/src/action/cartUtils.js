export const addItemToCart = (cartItems, cartItemToAdd) => {
  const existingCartItem = cartItems.find(
    cartItem => cartItem.id === cartItemToAdd.id
  );

  console.log(cartItemToAdd.id);

  if (existingCartItem) {
    return cartItems.map(cartItem =>
      cartItem.id === cartItemToAdd.id
        ? { ...cartItem, quantity: cartItem.quantity + 1 }
        : cartItem
    );
  }

  return [...cartItems, { ...cartItemToAdd, quantity: 1 }];
};

export const removeItemFromCart = (cartItems, cartItemToRemove) => {
  const existingCartItem = cartItems.find(
    cartItem => cartItem.id === cartItemToRemove.id
  );

  if (existingCartItem.quantity === 1) {
    return cartItems.filter(cartItem => cartItem.id !== cartItemToRemove.id);
  }
  return cartItems.map(cartItem =>
    cartItem.id === cartItemToRemove.id
      ? { ...cartItem, quantity: cartItem.quantity - 1 }
      : cartItem
  );
};

const amount = item => {
  return item.quantity;
};

const sum = price => {
  let sum = 0;
  return (sum += price);
};

export const getTotalAmount = cartItems => {
  /* this is our initial value i.e. the starting point*/
  const initialValue = 0;
  let finalValue;
  /* numbers array */
  const numbers = cartItems.map(item => item);

  const number2 = cartItems.map(item => item.price);
  /* reducer method that takes in the accumulator and next item */
  const reducer = (accumulator, item) => {
    return accumulator + item.quantity * item.price;
  };

  /* we give the reduce method our reducer function
  and our initial value */
  const total = numbers.reduce(reducer, initialValue);

  console.log(total);
  return total;
};
