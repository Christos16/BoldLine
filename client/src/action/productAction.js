import axios from 'axios';

import { GET_PRODUCTS, GET_PRODUCT, GET_CATEGORY } from './types';

export const getProducts = (variables, category) => async dispatch => {
  await axios
    .post(`/api/product/getProducts?department=${category}`, variables)
    .then(res => dispatch({ type: GET_PRODUCTS, payload: res.data }))
    .catch(err => console.log(err));
};

export const getProduct = productId => async dispatch => {
  await axios
    .get(`/api/product/products_by_id?id=${productId}&type=single`)
    .then(res => {
      dispatch({ type: GET_PRODUCT, payload: res.data[0] });
    })
    .catch(err => console.log(err));
};

export const getCategory = productId => dispatch => {
  axios
    .get(`/api/product/products_by_category?category=${productId}`)
    .then(res => {
      dispatch({ type: GET_CATEGORY, payload: res.data[0] });
    })
    .catch(err => console.log(err));
};
