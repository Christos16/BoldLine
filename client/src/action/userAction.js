import { SUCCESS_BUY } from './types';
import axios from 'axios';

export const transactionSuccess = (cartInfo, data) => {
  let variables = {
    cartInfo,
    paymentData: data
  };

  console.log(variables);
  axios
    .post('/api/users/successBuy', variables)
    .then(res => dispatch({ type: SUCCESS_BUY, payload: res.data }))
    .catch(err => console.log(err));
};
