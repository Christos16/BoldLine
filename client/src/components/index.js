export { default as SearchFeature } from './Sections/SearchFeature';
export { default as Checkbox } from './Sections/Checkbox';
export { default as Card } from './Card/CardItem';
export { default as RadioButton } from './Sections/RadioButton';
export { default as FilterSize } from './Sections/FilterSize';
export { default as Product } from './Product/Product';
export { default as Navbar } from './Layouts/Navbar';
export { default as Footer } from './Layouts/Footer';
export { default as MenCategory } from './landing/MenCategory';
export { default as WomenCategory } from './landing/WomenCategory';
export { default as FormInput } from './form-inputs/FormInput';
export { default as Loading } from './Layouts/Loading';
export { default as Cookies } from './Layouts/Cookies';
export { default as PrivacyPolicy } from './Layouts/Privacy-Policy';
