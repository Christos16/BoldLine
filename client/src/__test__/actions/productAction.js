import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock';
import configureMockStore from 'redux-mock-store';
import { GET_PRODUCTS } from '../../action/types';
import { getProducts } from '../../action/productAction';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('actions', () => {
  it('calls appropriate requests', function(done) {
    moxios.wait(() => {
      expect(moxios.requests.mostRecent().config.url).to.equal(
        'https://www.google.com'
      );
      moxios.wait(() => {
        expect(moxios.requests.mostRecent().config.url).to.equal(
          'https://www.yahoo.com'
        );
        done();
      });
    });
  });
});
