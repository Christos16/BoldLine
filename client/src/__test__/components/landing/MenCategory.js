import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import { findByTestAttr, testStore } from '../../../../utils/utils';
import MenCategory from '../../../components/landing/MenCategory';
import { Provider } from 'react-redux';
import * as ReactReduxHooks from '../../../../utils/react-redux-hooks';
import * as selectActions from '../../../action/productAction';
import toJson from 'enzyme-to-json';
const mockStore = configureStore();
const store = mockStore();
configure({ adapter: new Adapter() });

describe('MenCategory', () => {
  it('renders without crashing given the required props', () => {
    const props = {
      dispatch: jest.fn(),
      useSelector: jest.fn()
    };
    const wrapper = shallow(
      <Provider>
        <MenCategory {...props} />
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
