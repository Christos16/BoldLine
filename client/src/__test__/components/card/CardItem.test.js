import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import { findByTestAttr, testStore } from '../../../../utils/utils';
import MenCategory from '../../../components/landing/MenCategory';
import { Provider } from 'react-redux';
import * as ReactReduxHooks from '../../../../utils/react-redux-hooks';
import * as selectActions from '../../../action/productAction';
import toJson from 'enzyme-to-json';
import { Card } from '../../../components';
const mockStore = configureStore();

const store = mockStore();
configure({ adapter: new Adapter() });

describe('RecipeList', () => {
  let wrapper;
  let useEffect;
  let store;

  beforeEach(() => {
    store = configureStore()({
      products: [
        {
          size: 1,
          price: 25,
          menCategory: 1,
          womanCategory: 1,
          _id: '12345',
          imagePath: 'http://static.zara',
          title: 'Grey Pullover',
          description: 'Grey Pullover with pockets',
          color: 'Stone',
          quantity: 0,
          department: 'Men'
        },
        {
          size: 2,
          price: 22,
          menCategory: 2,
          womanCategory: 3,
          _id: '12345',
          imagePath: 'http://stfatic.zara',
          title: 'Grey ',
          description: ' pockets',
          color: 'red',
          quantity: 0,
          department: 'Men'
        }
      ]
    });

    const props = {
      product: {
        size: 1,
        price: 25,
        menCategory: 1,
        womanCategory: 1,
        _id: '12345',
        imagePath: 'http://static.zara',
        title: 'Grey Pullover',
        description: 'Grey Pullover with pockets',
        color: 'Stone',
        quantity: 0,
        department: 'Men'
      }
    };
    jest
      .spyOn(ReactReduxHooks, 'useSelector')
      .mockImplementation(state => store.getState());

    jest
      .spyOn(ReactReduxHooks, 'useDispatch')
      .mockImplementation(() => store.dispatch);

    wrapper = shallow(<Card store={store} {...props} />);
  });

  describe('MenCategory', () => {
    it('renders without crashing given the required props', () => {
      const props = {
        dispatch: jest.fn(),
        product: {
          size: 1,
          price: 25,
          menCategory: 1,
          womanCategory: 1,
          _id: '12345',
          imagePath: 'http://static.zara',
          title: 'Grey Pullover',
          description: 'Grey Pullover with pockets',
          color: 'Stone',
          quantity: 0,
          department: 'Men'
        },
        dispatch: jest.fn()
      };
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
