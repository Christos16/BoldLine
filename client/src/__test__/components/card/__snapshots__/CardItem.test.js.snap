// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList MenCategory renders without crashing given the required props 1`] = `undefined`;
